pkg/common/utils_test.go

```diff
@@ -234,36 +234,28 @@ func TestConvertSITimestamp(t *testing.T) {
 	assert.Equal(t, result, time.Time{})
 }
 
-func TestWaitFor(t *testing.T) {
-	var tests = []struct {
-		testname   string
-		bound      int
-		ErrorExist bool
+func TestWaitForCondition(t *testing.T) {
+	target := false
+	eval := func() bool {
+		return target
+	}
+	tests := []struct {
+		input    bool
+		interval time.Duration
+		timeout  time.Duration
+		output   error
 	}{
-		{"Timeout case", 10000, true},
-		{"Fullfilling case", 10, false},
+		{true, time.Duration(1) * time.Second, time.Duration(2) * time.Second, nil},
+		{false, time.Duration(1) * time.Second, time.Duration(2) * time.Second, ErrorTimeout},
```

Comment:
Could you please add test for "interval > timeout"? It should call `eval` only once